src/job.rs

```diff
@@ -184,19 +184,19 @@ impl Job {
                     // was sent we need to mark it in the database ASAP as we
                     // otherwise might send it twice.
                     let mut smtp = context.smtp.lock().unwrap();
-                    match smtp.send(context, recipients_list, body, self.job_id) {
-                        Err(SmtpError::SendError(err)) => {
+                    match smtp.send(context, recipients_list, body, format!("{}", self.job_id)) {
+                        Err(crate::smtp::sendmail::Error::SendError(err)) => {
                             // Remote error, retry later.
                             smtp.disconnect();
                             info!(context, "SMTP failed to send: {}", err);
                             self.try_again_later(TryAgain::AtOnce, Some(err.to_string()));
                         }
-                        Err(SmtpError::EnvelopeError(err)) => {
+                        Err(crate::smtp::sendmail::Error::EnvelopeError(err)) => {
                             // Local error, job is invalid, do not retry.
                             smtp.disconnect();
                             warn!(context, "SMTP job is invalid: {}", err);
                         }
-                        Err(SmtpError::NoTransport) => {
```

Comment:
This `NoTransport` error does not really makes sense as it is an internal error meaning that login parameters are invalid probably, but replacing it is not the point of this PR.